### PR TITLE
The engine does not eat errors anymore

### PR DESCRIPTION
**Pull request by micheles**
Solves https://bugs.launchpad.net/oq-engine/+bug/1469124.
The tests are running here: https://ci.openquake.org/job/zdevel_oq-engine/1301

### CODE REVIEW

File: openquake/engine/engine.py [hunk 1]
```diff
@@ -109,23 +109,34 @@ def job_stats(job):
...
         job.is_running = False
-        job.save()
-
-        # save job stats
-        curs.execute("select pg_database_size(%s)", (dbname,))
-        new_dbsize = curs.fetchall()[0][0]
-        js.disk_space = new_dbsize - dbsize
-        js.stop_time = datetime.utcnow()
-        js.save()
+        if tb != 'None\n':
```
Comment (daniviga) [inline openquake/engine/engine.py]: Just a curiosity, why newline is needed at the end?

File: openquake/engine/engine.py [hunk 2]
```diff
@@ -109,23 +109,34 @@ def job_stats(job):
...
+            # rollback the transactions; unfortunately, for mysterious reasons,
+            # this is not enough and an OperationError may still show up in the
+            # finalization phase when forks are involved
+            for conn in django_db.connections.all():
+                conn.rollback()
```
Comment (daniviga) [inline openquake/engine/engine.py]: Before `conn` was taken from `django_db.connections['job_init']`; now this has been removed. Where `conn` is from?

**Comment by daniviga** [comment]
LGTM

File: openquake/engine/engine.py (hunk 1 above)
Comment (micheles) [inline openquake/engine/engine.py]: I saw that `traceback.format_exc()` returns `None\n` when there is no traceback.

**Pull request merged**